doc/rtd/development/packaging.rst

```diff
@@ -0,0 +1,105 @@
+.. _building_and_distribution:
+
+Building and Distribution
+*************************
+
+``Cloud-init`` is distributed as a system package and pre-installed in cloud
+images. Correct behavior depends on the tight integration of system services
+performing system-wide configuration during early boot stages. Because of
+this, cloud-init is not published to PyPI as it is not intended to be consumed
+as a pure-python package. Instead, cloud-init uses meson_ to build all
+dependent data files, system-initialization scripts and global configuration
+files.
+
+The build system used by ``cloud-init`` is `meson`_. ``Cloud-init`` is not
+published to pypi.org because it is not intended to be consumed directly as a
+pure-python library, SDK or API for managing cloud configuration.
+
+``Cloud-init`` correct behavior depends on the tight integration of
+system services performing :ref:`the early boot stages<boot_stages>` and
+system-wide configuration. Upstream uses `meson`_ build backend to support the
+distribution of all dependent data files, system-initialization scripts and
+global configuration files.
+
+Guidelines
+==========
+
+Build Dependencies
+------------------
+To use ``meson``, the following build dependencies should be available on the
+system:
+
+- ``python3``
+- ``meson >= 0.63.0``
+- ``pkgconf``
+- ``bash-completion``
+
+Additional dependencies for systemd environments:
+
+- ``systemd-devel``
+- ``udev``
+
+Local build procedure
```

Comment:
I don't think that non-local building makes sense in the context of meson.

How about "Manual build proceedure"? Where "manual" implies that you are doing something that shouldn't normally need to be done.